src/main/studio/managers/ProjectTemplateCreator.js

```diff
@@ -0,0 +1,195 @@
+import * as dns from 'dns';
+import * as fs from 'fs';
+import { TEMPLATES } from '../../../contents';
+import { join } from 'path';
+import { app } from 'electron';
+import DecompressZip from 'decompress-zip';
+import ncp from 'ncp';
+import { promisify } from 'util';
+import rimraf from 'rimraf';
+
+/**
+ *
+ * @class ProjectTemplateCreator
+ */
+export default class ProjectTemplateCreator {
+  /**
+   * Creates an instance of ProjectTemplateCreator.
+   * @param {string} studio
+   * @memberof ProjectTemplateCreator
+   */
+  constructor(studio) {
+    this.studio = studio;
+    this.tempDir = join(app.getPath('userData'), 'tmp/');
+  }
+
+  /**
+   *
+   * @param {*} location
+   * @memberof ProjectTemplateCreator
+   */
+  isLocationEmpty(location) {
+    if (!fs.existsSync(location)) {
+      return true;
+    }
+    try {
+      const files = fs.readdirSync(location);
+      return files.length === 0;
+    }
+    catch (e) {
+      return false;
+    }
+  }
+
+  /**
+   *
+   * @param {string} type
+   * @param {string} location
+   * @returns
+   * @memberof ProjectTemplateCreator
+   */
+  create(type, location) {
+    return new Promise((resolve, reject) => {
+      if (!this.isLocationEmpty(location)) {
+        reject({ msg: 'New project location must be empty.' });
+        return;
+      }
+      dns.resolve('www.github.com', (err) => {
+        if (err) {
+          this.createFrom('file', type, location)
+            .then(resolve)
+            .catch(reject);
+        }
+        else {
+          this.createFrom('github', type, location)
+            .then(resolve)
+            .catch(reject);
+        }
+      });
+    });
+  }
+
+  /**
+   *
+   * @param {string} from
+   * @param {string} type
+   * @param {string} location
+   * @memberof ProjectTemplateCreator
+   */
+  createFrom(from, type, location) {
```

Comment:
This can be made async which I think will read better:

```js
async createFrom(from, type, location) {
  const path = await this.getTemplateZip(from, type);
  if (path === undefined) {
    throw new Error();
  }

  await this.extractTemplateFiles(path);
  await this.copyTemplateFilesTo(location);
}
```

Reply:
Fixed